x-pack/plugins/apm/public/components/app/service_dashboards/actions/save_dashboard_modal.tsx

```diff
@@ -0,0 +1,273 @@
+  EuiIcon,
+  EuiButtonEmpty,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { DashboardItem } from '@kbn/dashboard-plugin/common/content_management';
+import { callApmApi } from '../../../../services/rest/create_call_apm_api';
+import { useDashboardFetcher } from '../../../../hooks/use_dashboards_fetcher';
+import { FETCH_STATUS } from '../../../../hooks/use_fetcher';
+import { useApmPluginContext } from '../../../../context/apm_plugin/use_apm_plugin_context';
+import { useApmParams } from '../../../../hooks/use_apm_params';
+import { SERVICE_NAME } from '../../../../../common/es_fields/apm';
+import { MergedServiceDashboard } from '..';
+
+interface Props {
+  onClose: () => void;
+  onRefresh: () => void;
+  currentDashboard?: MergedServiceDashboard;
+  serviceDashboards?: MergedServiceDashboard[];
+}
+
+export function SaveDashboardModal({
+  onClose,
+  onRefresh,
+  currentDashboard,
+  serviceDashboards,
+}: Props) {
+  const {
+    core: { notifications },
+  } = useApmPluginContext();
+  const { data: allAvailableDashboards, status } = useDashboardFetcher();
+
+  let defaultOption: EuiComboBoxOptionOption<string> | undefined;
+
+  const [useServiceFilters, setUseServiceFilters] = useState(
+    currentDashboard?.useServiceFilters ?? true
+  );
+
+  if (currentDashboard) {
+    const { title, dashboardSavedObjectId } = currentDashboard;
+    defaultOption = { label: title, value: dashboardSavedObjectId };
+  }
+
+  const [selectedDashboard, setSelectedDashboard] = useState(
+    defaultOption ? [defaultOption] : []
+  );
+
+  const isEditMode = !!currentDashboard?.id;
+
+  const {
+    path: { serviceName },
+  } = useApmParams('/services/{serviceName}/dashboards');
+
+  const reloadServiceDashboards = useCallback(() => {
+    onRefresh();
+  }, [onRefresh]);
+
+  const options = allAvailableDashboards?.map(
+    (dashboardItem: DashboardItem) => ({
+      label: dashboardItem.attributes.title,
+      value: dashboardItem.id,
+      disabled:
+        serviceDashboards?.some(
+          ({ dashboardSavedObjectId }) =>
+            dashboardItem.id === dashboardSavedObjectId
+        ) ?? false,
+    })
+  );
+  const onSave = useCallback(
+    async function () {
+      const [newDashboard] = selectedDashboard;
+      try {
+        if (newDashboard.value) {
+          await callApmApi('POST /internal/apm/service-dashboard', {
+            params: {
+              query: { serviceDashboardId: currentDashboard?.id },
+              body: {
+                dashboardSavedObjectId: newDashboard.value,
+                useServiceFilters,
+                kuery: `${SERVICE_NAME}: ${serviceName}`,
+              },
+            },
+            signal: null,
+          });
+
+          notifications.toasts.addSuccess(
+            isEditMode
+              ? getEditSuccessToastLabels(newDashboard.label)
+              : getLinkSuccessToastLabels(newDashboard.label)
+          );
+          reloadServiceDashboards();
```

Comment:
As mentioned above, I don't think we should call it `serviceDashboards`. Instead we should call it `customDashboards` or `linkedDashboards`. 

```suggestion
          reloadServiceDashboards();
```